stake-pool/single-pool-cli/src/output.rs

```diff
@@ -0,0 +1,279 @@
+use {
+    crate::config::Config,
+    console::{style, Emoji},
+    serde::{Deserialize, Serialize},
+    serde_with::{serde_as, DisplayFromStr},
+    solana_cli_output::{display::writeln_name_value, OutputFormat, QuietDisplay, VerboseDisplay},
+    solana_sdk::native_token::Sol,
+    solana_sdk::{pubkey::Pubkey, signature::Signature, stake::state::Lockup},
+    spl_single_validator_pool::{
+        self as single_pool, find_pool_mint_address, find_pool_mint_authority_address,
+        find_pool_mpl_authority_address, find_pool_stake_address,
+        find_pool_stake_authority_address,
+    },
+    std::fmt::{Display, Formatter, Result, Write},
+};
+
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub(crate) struct CommandOutput<T>
+where
+    T: Serialize + Display + QuietDisplay + VerboseDisplay,
+{
+    pub(crate) command_name: String,
+    pub(crate) command_output: T,
+}
+
+impl<T> Display for CommandOutput<T>
+where
+    T: Serialize + Display + QuietDisplay + VerboseDisplay,
+{
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        Display::fmt(&self.command_output, f)
+    }
+}
+
+impl<T> QuietDisplay for CommandOutput<T>
+where
+    T: Serialize + Display + QuietDisplay + VerboseDisplay,
+{
+    fn write_str(&self, w: &mut dyn std::fmt::Write) -> std::fmt::Result {
+        QuietDisplay::write_str(&self.command_output, w)
+    }
+}
+
+impl<T> VerboseDisplay for CommandOutput<T>
+where
+    T: Serialize + Display + QuietDisplay + VerboseDisplay,
+{
+    fn write_str(&self, w: &mut dyn std::fmt::Write) -> std::fmt::Result {
+        writeln_name_value(w, "Command:", &self.command_name)?;
+        VerboseDisplay::write_str(&self.command_output, w)
+    }
+}
+
+pub fn format_output<T>(config: &Config, command_name: String, command_output: T) -> String
+where
+    T: Serialize + Display + QuietDisplay + VerboseDisplay,
+{
+    config.output_format.formatted_string(&CommandOutput {
+        command_name,
+        command_output,
+    })
+}
+
+#[serde_as]
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct SignatureOutput {
+    #[serde_as(as = "Option<DisplayFromStr>")]
+    pub signature: Option<Signature>,
+}
+
+impl QuietDisplay for SignatureOutput {}
+impl VerboseDisplay for SignatureOutput {}
+
+impl Display for SignatureOutput {
+    fn fmt(&self, f: &mut Formatter) -> Result {
+        writeln!(f)?;
+
+        if let Some(signature) = self.signature {
+            writeln_name_value(f, "Signature:", &signature.to_string())?;
+        }
+
+        Ok(())
+    }
+}
+
+#[serde_as]
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct StakePoolOutput {
+    #[serde_as(as = "DisplayFromStr")]
+    pub pool_address: Pubkey,
+    #[serde_as(as = "DisplayFromStr")]
+    pub vote_account_address: Pubkey,
+    pub available_stake: u64,
+    pub tokens_outstanding: u64,
```

Comment:
nit for clarity? or maybe `token_supply`?
```suggestion
    pub tokens_minted: u64,
```